temporian/core/operators/fft.py

```diff
@@ -0,0 +1,159 @@
+from temporian.proto import core_pb2 as pb
+from temporian.utils.rtcheck import rtcheck
+from temporian.core.data.dtype import DType
+
+
+_WINDOWS = {"hamming"}
+
+
+class FFT(Operator):
+    def __init__(
+        self,
+        input: EventSetNode,
+        num_events: int,
+        window: Optional[str] = None,
+        num_spectral_lines: Optional[int] = None,
+    ):
+        super().__init__()
+
+        # Save attributes
+        self._num_events = num_events
+        self._window = window
+        self._num_spectral_lines = num_spectral_lines
+
+        self.add_attribute("num_events", num_events)
+        if window is not None:
+            self.add_attribute("window", window)
+        if num_spectral_lines is not None:
+            self.add_attribute("num_spectral_lines", num_spectral_lines)
+
+        # Check errors
+        if num_events <= 1:
+            raise ValueError(
+                "num_events should be strictly positive. Got"
+                f" {num_events} instead."
+            )
+        if len(input.features) != 1:
+            raise ValueError(
+                "FFT input needs to be a single feature. Got"
+                f" {len(input.features)} features instead."
+            )
+        if input.features[0].dtype != DType.FLOAT32:
+            raise ValueError(
+                "FFT input feature should be tp.float32. Got"
+                f" {input.features[0].dtype} instead."
+            )
+        if window is not None and window not in _WINDOWS:
+            raise ValueError(
+                f"window should be None or in {_WINDOWS}. Got {window} instead."
+            )
+        if num_spectral_lines is not None:
+            if num_spectral_lines <= 0:
+                raise ValueError(
+                    "num_spectral_lines should be positive. Got"
+                    f" {num_spectral_lines} instead"
+                )
+            if num_spectral_lines > num_events // 2:
+                raise ValueError(
+                    "num_spectral_lines should be less or equal to num_events"
+                    f" // 2. Got {num_spectral_lines} instead"
+                )
+
+        self.add_input("input", input)
+        self.add_output(
+            "output",
+            create_node_new_features_new_sampling(
+                features=[
+                    (f"a{i}", DType.FLOAT32)
+                    for i in range(self.num_output_features)
+                ],
+                indexes=input.schema.indexes,
+                is_unix_timestamp=input.schema.is_unix_timestamp,
+                creator=self,
+            ),
+        )
+
+        self.check()
+
+    @property
+    def num_spectral_lines(self) -> Optional[int]:
+        return self._num_spectral_lines
+
+    @property
+    def num_output_features(self) -> int:
+        if self._num_spectral_lines is None:
+            return self._num_events // 2
+        else:
+            return self._num_spectral_lines
```

Comment:
Since "num_spectral_lines" is part of the API, user might expect it to have the same value as the "num_spectral_lines" constructor argument. This is why I keep them separated.